src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int sizeOfList;
+    private Node firstNode;
+    private Node lastNode;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node node = new Node(value);
+        if (sizeOfList == 0) {
+            firstNode = node;
+            lastNode = node;
+            sizeOfList += 1;
+            return;
+        }
+        lastNode.nextNode = node;
+        node.previousNode = lastNode;
+        lastNode = node;
+        sizeOfList += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == sizeOfList) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            addToStart(value, index);
+            return;
+        }
+        Node currentNextNode = getNodeByIndex(index);
+        Node currentPrevNode = currentNextNode.previousNode;
+        Node node = new Node(value);
+        currentPrevNode.nextNode = node;
+        node.previousNode = currentPrevNode;
+        node.nextNode = currentNextNode;
+        currentNextNode.previousNode = node;
```

Comment:
We can simplify it a bit:
```suggestion
        Node<T> node = new Node<>(value);
        Node<T> oldNode = getNodeByIndex(index);
        if (oldNode.previousNode != null) {
            node.previousNode = oldNode.previousNode;
            oldNode.previousNode.nextNode = node;
        } else {
            firstNode = node;
        }
        node.nextNode = oldNode;
        oldNode.previousNode = node;
```